src/engine/pieces/king.ts

```diff
@@ -1,13 +1,41 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import gameSettings from "../gameSettings";
+import Square from "../square";
 
 export default class King extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        const currentSquare = board.findPiece(this)
+        const possibleMoves: Square[] = []
+        if(currentSquare.row+1<gameSettings.BOARD_SIZE){
```

Comment:
This comparison to the board size happens quite a lot, which smells like an opportunity to commonise!